feature/amazon/src/test/java/com/revenuecat/purchases/amazon/handler/PurchaseUpdatesHandlerTest.kt

```diff
@@ -0,0 +1,217 @@
+import com.revenuecat.purchases.amazon.helpers.dummyUserData
+import io.mockk.mockk
+import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.api.Assertions.fail
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.util.Date
+
+@RunWith(AndroidJUnit4::class)
+class PurchaseUpdatesHandlerTest {
+
+    private lateinit var underTest: PurchaseUpdatesHandler
+    private val apiKey = "api_key"
+    private lateinit var cache: MockDeviceCache
+    private lateinit var purchasingServiceProvider: PurchasingServiceProviderForTest
+
+    private var receivedReceipts: List<Receipt>? = null
+    private var receivedUserData: UserData? = null
+    private var receivedError: PurchasesError? = null
+
+    private var expectedOnSuccess: (List<Receipt>, UserData) -> Unit = { receipts, userData ->
+        receivedReceipts = receipts
+        receivedUserData = userData
+    }
+    private var unexpectedOnSuccess: (List<Receipt>, UserData) -> Unit = { _, _ ->
+        fail("should be error")
+    }
+
+    private var expectedOnError: (PurchasesError) -> Unit = {
+        receivedError = it
+    }
+
+    private var unexpectedOnError: (PurchasesError) -> Unit = {
+        fail("should be success")
+    }
+
+    @Before
+    fun setup() {
+        cache = MockDeviceCache(mockk(), apiKey)
+        purchasingServiceProvider = PurchasingServiceProviderForTest()
+        underTest = PurchaseUpdatesHandler(purchasingServiceProvider)
+    }
+
+    @Test
+    fun `successfully queries purchases`() {
+        val dummyRequestId = "a_request_id"
+        purchasingServiceProvider.getPurchaseUpdatesRequestId = dummyRequestId
+
+        underTest.queryPurchases(
+            expectedOnSuccess,
+            unexpectedOnError
+        )
+
+        assertThat(purchasingServiceProvider.getPurchaseUpdatesCalled).isTrue
+
+        val receipts = listOf(getReceipt("receipt_id_1"), getReceipt("receipt_id_2"))
+        val response = getDummyPurchaseUpdatesResponse(dummyRequestId)
+
+        underTest.onPurchaseUpdatesResponse(response)
+
+        assertThat(receivedReceipts).isNotNull
+        assertThat(receivedReceipts).isNotEmpty
+        assertThat(receivedReceipts).hasSameSizeAs(receipts)
+
+        receipts.forEach { expected ->
+            val filteredReceipt = receivedReceipts!!.filter {
+                it.isCanceled == expected.isCanceled &&
+                    it.receiptId == expected.receiptId &&
+                    it.cancelDate == expected.cancelDate &&
+                    it.productType == expected.productType &&
+                    it.purchaseDate == expected.purchaseDate &&
+                    it.sku == expected.sku
+            }
+            assertThat(filteredReceipt).isNotNull
+        }
+
+        assertThat(receivedUserData).isEqualTo(response.userData)
+    }
+
+    @Test
+    fun `successfully queries purchases and there are no receipts`() {
```

Comment:
`if there are no receipts when successfully querying purchases, it returns an emtpy list`